Frame.cc

```diff
@@ -804,10 +810,13 @@ bool Frame::fillSpectralProfileData(int regionId, CARTA::SpectralProfileData& pr
                     auto cursorPos = region->getControlPoints()[0];
                     // try use the loader's optimized cursor profile reader first
                     if (!loader->getCursorSpectralData(spectralData, profileStokes, cursorPos.x(), cursorPos.y())) {
-                        getSpectralData(spectralData, sublattice, 100);
+                        // get spectral profile data
+                        bool complete = getSpectralData(spectralData, sublattice, 100);
+                        guard.unlock();
+                        // only send spectral profile data to frontend while it is complete
+                        if (complete)
+                            region->fillSpectralProfileData(profileData, i, spectralData);
```

Comment:
The job count is decremented at the end of getSpectralData(), Frame could be deleted while its Region is filling the spectral profile message.

Reply:
Moving the count increment/decrement to the function `Frame::fillSpectralProfileData()` to make sure the whole spectral profile process is done before deleting the Frame.